lib/ddtrace/transport/http/adapters/net.rb

```diff
@@ -16,9 +16,10 @@ class Net
 
           DEFAULT_TIMEOUT = 30
 
-          def initialize(hostname, port, options = {})
-            @hostname = hostname
-            @port = port
+          # @deprecated Positional parameters are deprecated. Use named parameters instead.
+          def initialize(hostname = nil, port = nil, **options)
+            @hostname = hostname || options[:hostname]
+            @port = port || options[:port]
```

Comment:
Minor: Should we use `options.fetch(:hostname)` and `options.fetch(:port)` here?

(same note for similar changes in other adapters; this will not apply if we end up just passing the agent settings object directly)